rust/bridge/shared/src/jni/convert.rs

```diff
@@ -188,6 +189,36 @@ impl<'a> SimpleArgTypeInfo<'a> for Option<String> {
     }
 }
 
+impl<'a> SimpleArgTypeInfo<'a> for Uuid {
+    type ArgType = JObject<'a>;
+    fn convert_from(env: &JNIEnv, foreign: JObject<'a>) -> SignalJniResult<Self> {
+        check_jobject_type(env, foreign, "java/util/UUID")?;
+        let msb = match call_method_checked(env, foreign, "getMostSignificantBits", "()J", &[])? {
+            JValue::Long(value) => value,
+            anything_else => {
+                return Err(SignalJniError::UnexpectedJniResultType(
+                    "getMostSignificantBits",
+                    anything_else.type_name(),
+                ))
+            }
+        };
+        let lsb = match call_method_checked(env, foreign, "getLeastSignificantBits", "()J", &[])? {
+            JValue::Long(value) => value,
+            anything_else => {
+                return Err(SignalJniError::UnexpectedJniResultType(
+                    "getLeastSignificantBits",
+                    anything_else.type_name(),
+                ))
+            }
+        };
```

Comment:
Lots of code duplication here feels like it could be an inner fn or a loop

Reply:
Yeah, `call_method_checked` can probably take care of this! \*goes off to implement\*

Reply:
Okay, how do you feel about the latest commit?

Reply:
Looks great, checking the return type removes a lot of noise.